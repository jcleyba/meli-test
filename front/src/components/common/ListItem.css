.list-item {
  display: flex;
  padding: 16px 0;
  border-bottom: solid 1px #eee;
  justify-content: flex-start; }
  .list-item:first-of-type {
    padding-top: 0; }
  .list-item:last-child {
    padding-bottom: 0;
    border-bottom: 0; }
  .list-item .thumbnail {
    margin-right: 16px; }
    .list-item .thumbnail img {
      border-radius: 4px;
      width: 180px;
      height: 180px; }
  .list-item .price {
    display: flex;
    align-items: center;
    margin-bottom: 32px;
    font-size: 24px;
    margin-top: 16px; }
    .list-item .price .free-shipping {
      margin-left: 10px; }
      .list-item .price .free-shipping img {
        max-width: 30px; }
  .list-item .description {
    flex: 3; }
  .list-item .title {
    font-size: 18px; }
    .list-item .title a {
      text-decoration: none;
      color: black; }
  .list-item .location {
    flex: 1;
    justify-self: flex-end;
    color: #999;
    font-size: 12px; }
