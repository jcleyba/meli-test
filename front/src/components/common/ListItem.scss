@import "../../styles/_variables";

.list-item {
  display: flex;
  padding: $box_padding 0;
  border-bottom: solid 1px $light_grey;
  justify-content: flex-start;

  &:first-of-type {
    padding-top: 0;
  }

  &:last-child {
    padding-bottom: 0;
    border-bottom: 0;
  }
  .thumbnail {
    margin-right: $box_padding;
    img {
      border-radius: $thumb_border_radius;
      width: 180px;
      height: 180px;
    }
  }
  .price {
    display: flex;
    align-items: center;
    margin-bottom: $price_margin;
    font-size: $price_font_size;
    margin-top: $box_padding;
    .free-shipping {
      margin-left: 10px;

      img {
        max-width: 30px;
      }
    }
  }

  .description {
    flex: 3;
  }

  .title {
    font-size: $title_font_size;
    a {
      text-decoration: none;
      color: black;
    }
  }

  .location {
    flex: 1;
    justify-self: flex-end;
    color: $grey;
    font-size: $location_font_size;
  }
}
