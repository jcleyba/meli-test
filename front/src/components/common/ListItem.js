import React from 'react';
import { NavLink } from 'react-router-dom';
import ProgressiveImage from 'react-progressive-image'

import shipping from '../../images/ic_shipping@2x.png.png';
import './ListItem.css';

const ListItem = (props) => {
  const renderThumb = (thumb, title) => {
    const picURI = thumb && thumb.replace('I', 'N');

    // Using progressive image for better UX on loading of images.
    return <ProgressiveImage src={picURI} placeholder={thumb}>
      {(src) => <img src={src} alt={title} title={title} className="product-thumb"/>}
    </ProgressiveImage>
  };

  const renderPrice = (price) => {
    return <span className="amount">$ {price.amount}
    </span>
  };

  const renderShipping = (freeShipping) => {
    return freeShipping && <span className="free-shipping"><img src={shipping} alt="shipping"/></span>

  };

  const renderCard = (itemData) => {
    const {id, title, price, thumbnail, free_shipping, address} = itemData;
    return (
      <div className="list-item">
        <div className="thumbnail">
          <NavLink to={`/items/${id}`}>{renderThumb(thumbnail, title)}</NavLink>
        </div>
        <div className="description">
          <div className="price">
            {renderPrice(price)}
            {renderShipping(free_shipping)}
          </div>
          <div className="title">
            <NavLink to={`/items/${id}`}>{title}</NavLink>
          </div>
        </div>

        <div className="location">
          {address.state_name}
        </div>
      </div>
    )
  };

  return renderCard(props.itemData);
};

export default ListItem;
