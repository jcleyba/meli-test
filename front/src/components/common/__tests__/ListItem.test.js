import React from 'react';
import { shallow, mount } from 'enzyme';
import 'jest-enzyme';
import { MemoryRouter } from 'react-router-dom';

import ListItem from '../ListItem';

describe('List Item Component', () => {

  let wrapper;
  let mounted;

  const props = {
    itemData: {
      id: 1,
      title: 'Name',
      price: {
        amount: 10
      },
      address: {
        state_name: 'Cordoba'
      }
    },
  };


  const Component = (props) => {
    return <MemoryRouter>
      <ListItem {...props}/>
    </MemoryRouter>
  };


  beforeEach(() => {
    wrapper = shallow(<Component {...props}/>);
    mounted = mount(<Component {...props}/>);
  });

  afterEach(() => {
    wrapper = null;
    mounted = null;
  });

  describe('List Item', () => {

    it('matches current snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should have itemData prop', () => {
      expect(mounted).toHaveProp('itemData');
    });

    it('should have a breadcrumbs className', () => {
      expect(mounted.find('.list-item')).toBePresent();
    });

    it('should have title same as prop', () => {
      expect(mounted.find('.title a').text()).toEqual(props.itemData.title);
    });

    it('should render a title as link', () => {
      expect(mounted.find('.title a')).toBePresent();
    });

    it('should render a thumbnail as link', () => {
      expect(mounted.find('.thumbnail a')).toBePresent();
    });

    it('should have a thumbnail className', () => {
      expect(mounted.find('.thumbnail')).toBePresent();
    });

    it('should render a description', () => {
      expect(mounted.find('.description')).toBePresent();
    });

    it('should render the name of location', () => {
      expect(mounted.find('.location').text()).toEqual(props.itemData.address.state_name);
    });
  });
})
;
