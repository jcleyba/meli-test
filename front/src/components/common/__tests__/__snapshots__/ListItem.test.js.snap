// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item Component List Item matches current snapshot 1`] = `"<div class=\\"list-item\\"><div class=\\"thumbnail\\"><a aria-current=\\"false\\" href=\\"/items/1\\"><img alt=\\"Name\\" title=\\"Name\\" class=\\"product-thumb\\"/></a></div><div class=\\"description\\"><div class=\\"price\\"><span class=\\"amount\\">$ 10</span></div><div class=\\"title\\"><a aria-current=\\"false\\" href=\\"/items/1\\">Name</a></div></div><div class=\\"location\\">Cordoba</div></div>"`;
